src/CharacterState.js

```diff
@@ -122,7 +209,111 @@ export default class CharacterState {
-            this.path
+            this.path,
+            this.sceneDimensions
+        );
+    }
+
+    moveUpPosition(): CharacterState {
+        let yPos = 0;
+        if (this.sceneDimensions.getBoundsStateY(this.yPos - 1) === 'outOfBoundsBelow') {
+            yPos = 1;
+        } else {
+            yPos = this.yPos - 1;
+        }
+        return new CharacterState(
+            this.xPos,
+            yPos,
+            this.direction,
+            this.path,
+            this.sceneDimensions
+        );
+    }
+
+    moveRightPosition(): CharacterState {
+        let xPos = 0;
+        if (this.sceneDimensions.getBoundsStateX(this.xPos + 1) === 'outOfBoundsAbove') {
+            xPos = this.sceneDimensions.getWidth();
+        } else {
+            xPos = this.xPos + 1;
+        }
+        return new CharacterState(
+            xPos,
+            this.yPos,
+            this.direction,
+            this.path,
+            this.sceneDimensions
         );
     }
+
+    moveDownPosition(): CharacterState {
+        let yPos = 0;
+        if (this.sceneDimensions.getBoundsStateY(this.yPos + 1) === 'outOfBoundsAbove') {
+            yPos = this.sceneDimensions.getHeight();
+        } else {
+            yPos = this.yPos + 1;
+        }
+        return new CharacterState(
+            this.xPos,
+            yPos,
+            this.direction,
+            this.path,
+            this.sceneDimensions
+        );
+    }
+
+    moveLeftPosition(): CharacterState {
+        let xPos = 0;
+        if (this.sceneDimensions.getBoundsStateY(this.xPos - 1) === 'outOfBoundsBelow') {
+            xPos = 1;
+        } else {
+            xPos = this.xPos - 1;
+        }
+        return new CharacterState(
+            xPos,
+            this.yPos,
+            this.direction,
+            this.path,
+            this.sceneDimensions
+        );
+    }
+
+    changeXPosition(columnLabel: string): CharacterState {
+        let newXPos = this.xPos;
+        if (columnLabel <= String.fromCharCode(64 + this.sceneDimensions.getWidth()) && columnLabel >='A') {
+            newXPos = columnLabel.charCodeAt(0) - 64;
+        } else if (columnLabel <= String.fromCharCode(96 + this.sceneDimensions.getWidth()) && columnLabel >='a') {
+            newXPos = columnLabel.charCodeAt(0) - 96;
+        }
+        return new CharacterState(
+            newXPos,
+            this.yPos,
+            this.direction,
+            this.path,
+            this.sceneDimensions
+        );
+    }
+
+    changeYPosition(rowLabel: number): CharacterState {
+        let newYPos = this.yPos;
+        if (rowLabel <= this.sceneDimensions.getHeight() && rowLabel >= 1) {
+            newYPos = rowLabel;
+        }
+        return new CharacterState(
+            this.xPos,
+            newYPos,
+            this.direction,
+            this.path,
+            this.sceneDimensions
+        );
+    }
+
+
+    getRowLabel(): string {
```

Comment:
I'm a little uneasy having the type `string` for the return values here.  It seems like we could protect ourselves better by enumerating the valid values in `types.js`.

Reply:
Currently, expected row label values are from 1 to 16, and this may change between any values from 1 to 26, based on our decision on how many rows we want in the scene. Specifying all those values as a type might be undesirable.

Reply:
As long as it's clearly identified as providing a string to be used as a label, it's fine.  We can talk about adding more checks if we need them to help avoid actual issues.